Saule/JsonApiSerializerOfT.cs

```diff
@@ -91,6 +91,18 @@ public JToken Serialize(object @object, Uri requestUri)
             return JsonApiSerializer.Serialize(preprocessResult);
         }
 
+        /// <summary>
+        /// Converts json into an object of a specified type
+        /// </summary>
+        /// <param name="object">Json to convert</param>
+        /// <param name="type">Type to convert to</param>
+        /// <returns>Json converted into the specified type of object</returns>
+        public dynamic Deserialize(JToken @object, Type type)
```

Comment:
Not a huge fan of `dynamic` here, but I'm assuming you have a good reason. I propose one of the following, but if that wouldn't work we can re-evaluate:

1. Change the signature to `public TResult Deserialize<T>(JToken)`
2. Change the signature to `public object Deserialize(JToken, Type)`

Alternatively we can have an overload for both.

Reply:
Thought about using option 2, but then I had to make a cast in the unit test/where I use the method.

i.e.
`var dsPerson = (Person)target.Deserialize(personJson, typeof(Person));
...dsPerson.FirstName;`

instead of just
`var dsPerson = target.Deserialize(personJson, typeof(Person));
...dsPerson.FirstName;`

That's why I preferred using dynamic.

Reply:
It's fine by me to do a cast in the test, as long as it's likely that the consumers of the API will also do a cast. `dynamic` adds some overhead and feels a bit 'unconventional', so if that was the only reason I prefer not to do it.

Reply:
Ok. I committed a return `object` version.